src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,136 @@
 import { PeopleFilters } from './PeopleFilters';
-import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { useSearchParams } from 'react-router-dom';
+import { Loader } from './Loader';
+import { getPeople } from '../api';
+
+function getExtendedPersonList(data: Person[]): Person[] {
+  return data.map(person => ({
+    ...person,
+    mother: data.find(curPerson => person.motherName === curPerson.name),
+    father: data.find(curPerson => person.fatherName === curPerson.name),
+  }));
+}
+
+function getFilteredPersonList(
+  data: Person[],
+  query: string | null,
+  filterStatus: string,
+  centuries: string[],
+  sort: string,
+  order: string,
+) {
+  let resultData = [...data];
+
+  if (filterStatus) {
+    resultData = resultData.filter(person => person.sex === filterStatus);
+  }
+
+  if (query) {
+    resultData = resultData.filter(person => {
+      const { name, motherName, fatherName } = person;
+      const lowerQuery = query.toLocaleLowerCase();
+      const nameMatches = name.toLocaleLowerCase().includes(lowerQuery);
+      const motherNameMatches =
+        motherName && motherName.toLocaleLowerCase().includes(lowerQuery);
+      const fatherNameMatches =
+        fatherName && fatherName.toLocaleLowerCase().includes(lowerQuery);
```

Comment:
`.toLocaleLowerCase().includes(lowerQuery)` you do it so many times, create a function